owner-tool/src/main.rs

```diff
@@ -797,55 +797,69 @@ async fn report_to_rendezvous(matches: &ArgMatches<'_>) -> Result<(), Error> {
     let owner_addresses: Result<Vec<Vec<TO2AddressEntry>>> =
         owner_addresses.drain(..).map(|v| v.try_into()).collect();
     let mut owner_addresses = owner_addresses.context("Error parsing owner addresses")?;
-    let owner_addresses = owner_addresses.drain(..).flatten().collect();
+    let owner_addresses: Vec<TO2AddressEntry> = owner_addresses.drain(..).flatten().collect();
 
     // Determine the RV IP
-    let rv_info = ov
-        .get_header()
-        .context("Error getting OV header")?
+    let rv_info = ov_header
         .rendezvous_info
         .to_interpreted(RendezvousInterpreterSide::Owner)
         .context("Error parsing rendezvous directives")?;
     if rv_info.is_empty() {
         bail!("No rendezvous information found that's usable for the owner");
     }
-    // Use the first entry
-    let rv_info = rv_info.first().unwrap();
-    let rv_urls = rv_info.get_urls();
-    if rv_urls.is_empty() {
-        bail!("No usable rendezvous URLs were found");
-    }
-    let rv_url = rv_urls.first().unwrap();
-
-    println!("Using rendezvous server at url {}", rv_url);
+    let mut rendezvous_performed = false;
+    for rv_directive in rv_info {
+        let rv_urls = rv_directive.get_urls();
+        if rv_urls.is_empty() {
+            bail!("No usable rendezvous URLs were found");
```

Comment:
I think that we might want to just continue if we don’t get useful URLs. It could be that that specific entry only contained device-specific items.

Reply:
Log an error/trace and continue to the next entry? Does that sound good?

Reply:
An `info` I'd say, since it's not an error.